hazelcast/src/main/java/com/hazelcast/instance/HazelcastInstanceFactory.java

```diff
@@ -29,4 +29,6 @@ HazelcastInstance newHazelcastInstance(Config config, String instanceName,
                                            NodeContext nodeContext) throws Exception;
 
     HazelcastInstance newHazelcastInstance(Config config) throws Exception;
+
+    HazelcastInstanceProxy newHazelcastInstanceProxy(HazelcastInstanceImpl hazelcastInstance);
```

Comment:
Is there another way rather than using `HazelcastInstanceImpl` because AFAIK it is in private API?